stm/asyncTaskQueue.mli

```diff
@@ -6,79 +6,129 @@
 (*         *       GNU Lesser General Public License Version 2.1        *)
 (************************************************************************)
 
+(** A status is
+    - `Fresh when XXX
+    - `Old x when XXX
+  *)
 type 'a worker_status = [ `Fresh | `Old of 'a ]
 
+(** The [Task] module type defines an abstract message-processing
+    entity. *)
 module type Task = sig
 
+  (** Identifier of a task, for workers that can process several
+      jobs. Usually an integer or any other unique identifier. *)
```

Comment:
A task is a lightweight description of a task for a worker, i.e. it could be "compute from state 15 to 34", and only contains "15" and "34".  It is not required to be marshalable.

Reply:
This is not a task id in the sense of your following comments.

Reply:
Would it be accurate to call it the task internal state?

Reply:
No, it is not the state, it is a description of it.